target/linux/realtek/files-6.12/drivers/net/phy/rtl83xx-phy.c

```diff
@@ -0,0 +1,4069 @@
+{
+	if (soc_info.family != RTL9300_FAMILY_ID)
+		return -ENODEV;
+
+	phydev_info(phydev, "Detected internal RTL9300 Serdes\n");
+
+	return 0;
+}
+
+static struct phy_driver rtl83xx_phy_driver[] = {
+	{
+		PHY_ID_MATCH_EXACT(PHY_ID_RTL8214C),
+		.name		= "Realtek RTL8214C",
+		.features	= PHY_GBIT_FEATURES,
+		.probe		= rtl8214c_phy_probe,
+		.read_page	= rtl821x_read_page,
+		.write_page	= rtl821x_write_page,
+		.suspend	= genphy_suspend,
+		.resume		= genphy_resume,
+	},
+	{
+		.match_phy_device = rtl8214fc_match_phy_device,
+		.name		= "Realtek RTL8214FC",
+		.config_aneg	= rtl8214fc_config_aneg,
+		.get_eee	= rtl8214fc_get_eee,
+		.get_features	= rtl8214fc_get_features,
+		.get_tunable    = rtl8214fc_get_tunable,
+		.probe		= rtl8214fc_phy_probe,
+		.read_page	= rtl821x_read_page,
+		.read_status    = rtl8214fc_read_status,
+		.resume		= rtl8214fc_resume,
+		.set_eee	= rtl8214fc_set_eee,
+		.set_tunable	= rtl8214fc_set_tunable,
+		.suspend	= rtl8214fc_suspend,
+		.write_page	= rtl821x_write_page,
+	},
+	{
+		.match_phy_device = rtl8218b_ext_match_phy_device,
+		.name		= "Realtek RTL8218B (external)",
+		.features	= PHY_GBIT_FEATURES,
+		.probe		= rtl8218b_ext_phy_probe,
+		.read_page	= rtl821x_read_page,
+		.write_page	= rtl821x_write_page,
+		.suspend	= genphy_suspend,
+		.resume		= genphy_resume,
+		.set_eee	= rtl8218b_set_eee,
+		.get_eee	= rtl8218b_get_eee,
+	},
+	{
+		PHY_ID_MATCH_EXACT(PHY_ID_RTL8218D),
+		.name		= "REALTEK RTL8218D",
+		.features	= PHY_GBIT_FEATURES,
+		.probe		= rtl8218d_phy_probe,
+		.read_page	= rtl821x_read_page,
+		.write_page	= rtl821x_write_page,
+		.suspend	= genphy_suspend,
+		.resume		= genphy_resume,
+		.set_eee	= rtl8218d_set_eee,
+		.get_eee	= rtl8218d_get_eee,
+	},
+	{
+		PHY_ID_MATCH_MODEL(PHY_ID_RTL8221B),
+		.name           = "REALTEK RTL8221B",
+		.features       = PHY_GBIT_FEATURES,
+		.suspend        = genphy_suspend,
+		.resume         = genphy_resume,
+		.read_page      = rtl821x_read_page,
+		.write_page     = rtl821x_write_page,
+		.read_status    = rtl8226_read_status,
+		.config_aneg    = rtl8226_config_aneg,
+		.set_eee        = rtl8226_set_eee,
+		.get_eee        = rtl8226_get_eee,
+	},
+	{
+		PHY_ID_MATCH_MODEL(PHY_ID_RTL8226),
+		.name		= "REALTEK RTL8226",
+		.features	= PHY_GBIT_FEATURES,
+		.suspend	= genphy_suspend,
+		.resume		= genphy_resume,
+		.read_page	= rtl821x_read_page,
+		.write_page	= rtl821x_write_page,
+		.read_status	= rtl8226_read_status,
+		.config_aneg	= rtl8226_config_aneg,
+		.set_eee	= rtl8226_set_eee,
+		.get_eee	= rtl8226_get_eee,
+	},
+	{
+		PHY_ID_MATCH_MODEL(PHY_ID_RTL8218B_I),
+		.name		= "Realtek RTL8218B (internal)",
+		.features	= PHY_GBIT_FEATURES,
+		.probe		= rtl8218b_int_phy_probe,
+		.read_page	= rtl821x_read_page,
+		.write_page	= rtl821x_write_page,
+		.suspend	= genphy_suspend,
+		.resume		= genphy_resume,
+		.set_eee	= rtl8218b_set_eee,
+		.get_eee	= rtl8218b_get_eee,
+	},
+	{
+		PHY_ID_MATCH_MODEL(PHY_ID_RTL8218B_I),
```

Comment:
Are you sure the `PHY_ID_RTL8218B_I` in this line is correct? The same value is also used for the preceding `PHY_ID_MATCH_MODEL()` section line 4002.